### PR TITLE
chore: drop order dependency from Data.Nat.Defs

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
There's an import chain `Order.Nat` → `Order.BoundedOrder` → `Order.Lattice` → { `Order.Monotone.Basic`, `Order.RelClasses` } which means none of these can use the `LinearOrder` instance anymore.

---

- [x] depends on: #13082 

I tried to limit the scope as much as possible here; there's more in https://github.com/leanprover-community/mathlib4/compare/Nat-LinearOrder-2 and https://github.com/leanprover-community/mathlib4/compare/List.Basic-no-algebra

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by YaelDillies** [comment]
I must say I've toyed with the idea myself but I'm not yet sure I like it.

The lemmas you're removing the dependencies on are very basic and boil down to easy-but-annoying-to-prove-on-the-fly tautologies.

I suspect the correct solution is to upstream those lemmas to Batteries using Batteries-specific typeclasses

**Comment by Ruben-VandeVelde** [comment]
That's certainly not a project I want to start on :)

File: Mathlib/Data/Nat/Defs.lean [hunk 1]
```diff
@@ -343,7 +330,7 @@ lemma add_succ_lt_add (hab : a < b) (hcd : c < d) : a + c + 1 < b + d := by
 #align nat.add_succ_lt_add Nat.add_succ_lt_add
 
 theorem le_or_le_of_add_eq_add_pred (h : a + c = b + d - 1) : b ≤ a ∨ d ≤ c := by
-  rcases le_or_lt b a with h' | h' <;> [left; right]
+  rcases (Nat.lt_or_ge a b).symm with h' | h' <;> [left; right]
```
Comment (YaelDillies) [inline Mathlib/Data/Nat/Defs.lean:333]: Can you add that as a file? Better let users not play the game of "Does this lemma have a `Nat` version or should I be reproving it?"

File: Mathlib/Data/Nat/Defs.lean [hunk 2]
```diff
@@ -1860,23 +1852,24 @@ lemma findGreatest_mono_right (P : ℕ → Prop) [DecidablePred P] {m n} (hmn :
...
-  · exact le_trans ih $ le_trans (findGreatest_le _) (le_succ _)
+  · exact Nat.le_trans ih $ Nat.le_trans (findGreatest_le _) (le_succ _)
   · exact ih
 #align nat.find_greatest_mono_right Nat.findGreatest_mono_right
 
 lemma findGreatest_mono_left [DecidablePred Q] (hPQ : ∀ n, P n → Q n) (n : ℕ) :
     Nat.findGreatest P n ≤ Nat.findGreatest Q n := by
   induction' n with n hn
-  · rfl
+  · exact Nat.le_refl _
```
Comment (YaelDillies) [inline Mathlib/Data/Nat/Defs.lean:1862]: Shouldn't `Nat.le_rfl` be a rfl lemma?

File: Mathlib/Data/Nat/SuccPred.lean [hunk 3]
```diff
@@ -3,6 +3,7 @@ Copyright (c) 2021 Yaël Dillies. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Yaël Dillies
 -/
+import Mathlib.Algebra.Order.Group.Nat
 import Mathlib.Algebra.Order.Ring.Nat
```
Comment (YaelDillies) [inline Mathlib/Data/Nat/SuccPred.lean:7]: The first import is redundant

**Review by YaelDillies — COMMENTED** [review]
I'm starting to wonder whether we're not past the useful import minimisation here. I will write more about it on Zulip tomorrow (please remind me tomorrow 8pm if I forget).

File: Mathlib/Data/Nat/SuccPred.lean (hunk 3 above)
Comment (Ruben-VandeVelde) [reply Mathlib/Data/Nat/SuccPred.lean:7]: It doesn't seem to be

**Comment by Ruben-VandeVelde** [comment]
I still think this is useful; it's a blocker for `Data.List.Basic`, among other things.

File: Mathlib/Init/Order/Defs.lean [hunk 4]
```diff
@@ -9,6 +9,8 @@ import Mathlib.Init.Data.Ordering.Basic
 import Mathlib.Tactic.SplitIfs
 import Mathlib.Tactic.TypeStar
 import Batteries.Classes.Order
+import Mathlib.Tactic.GCongr.Core
+
```
Comment (YaelDillies) [inline Mathlib/Init/Order/Defs.lean:13]:
```suggestion
```

**Review by YaelDillies — COMMENTED** [review]
What I am saying is that maybe we should allow basic files to depend on (a version of) `Preorder`/`PartialOrder`/`LinearOrder`. After all,
* Typeclasses were invented to avoid duplicating many lemmas. We are exactly going the other way around here.
* The reason we want to avoid basic algebra/order theory is to avoid almost horizontal dependencies between them, but we could add a "foundational" layer to the library containing `Foundation.PartialOrder`, `Foundation.LinearOrder`, `Foundation.Monoid`, `Foundation.AddMonoid`, `Foundation.Group`, `Foundation.AddGroup` (+ a few compatibility classes) so that
  * foundational algebra is used in basic algebra
  * foundational algebra is used in basic order theory
  * foundational order theory is used in basic algebra
  * foundational order theory is used in basic order theory
  
  but no other cross import is allowed.

**Pull request closed without merging**